sospice/catalog/catalog.py

```diff
@@ -451,3 +453,60 @@ def plot_fov(self, ax, **kwargs):
                 fontsize="xx-large",
                 borderaxespad=2,
             )
+
+    def download_files(
+        self,
+        base_dir,
+        base_url=None,
+        keep_tree=True,
+        downloader=None,
+        max_download=1000,
+    ):
+        """
+        Download all files from Catalog.,
+
+        Parameters
+        ----------
+        base_dir: Path or str
+            Base directory to download file to
+        base_url: str
+            Base URL for file
+        keep_tree: bool
+            Keep tree directory structure (by level and date)
+        downloader: parfive.Downloader
+            If provided, enqueue file for download instead of downloading it.
+            To download enqueued files, run `downloader.download()`
+        max_download: int
+            default maximum of 1000 files can be downloaded.
+            User can override it by changing the value.
+
+        Return
+        ------
+        parfive.Result
+            Download result (or None if file has only been enqueued)
+        """
+        get_catalog = self.read_catalog()
```

Comment:
so that, means it should reference the files directly from self obj

Reply:
`self.dataframe` like this

Reply:
The catalog is `self` (a `Catalog` object), which derives from a `pd.DataFrame` (although this might change in the future, see #42), so it should be sufficient to do `self.iloc[:max_download].apply(...)`, no need for `get_catalog` or `self.dataframe`.

Reply:
got it .